### PR TITLE
Remove StreamableReader

### PR DESCRIPTION
**Pull request by nik9000**
Relates to #17085

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java [hunk 1]
```diff
@@ -31,20 +31,19 @@
...
  */
 public interface Writeable<T> { // TODO remove <T>
     /**
      * Write this into the {@linkplain StreamOutput}.
      */
     void writeTo(StreamOutput out) throws IOException;
 
+    /**
+     * Read this object from a stream. Use a {@link Writeable.Reader} instead. This lives on for backwards compatibility but should be
+     * removed before 5.0.0GA. It is not deprecated because Diffable extends this interface and it shouldn't deprecated there.
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: shouldn't *be* deprecated

Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: ff730eb

File: core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java [hunk 2]
```diff
@@ -31,20 +31,19 @@
...
  */
 public interface Writeable<T> { // TODO remove <T>
     /**
      * Write this into the {@linkplain StreamOutput}.
      */
     void writeTo(StreamOutput out) throws IOException;
 
+    /**
+     * Read this object from a stream. Use a {@link Writeable.Reader} instead. This lives on for backwards compatibility but should be
+     * removed before 5.0.0GA. It is not deprecated because Diffable extends this interface and it be shouldn't deprecated there.
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java]: Nit: Almost ;-) To be or not to be, and also where, that is the question.

**Comment by cbuescher** [comment]
LGTM, left one tiny comment.

**Pull request merged**